guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/RangeDeserializer.java

```diff
@@ -188,9 +216,50 @@ public Range<?> deserialize(JsonParser p, DeserializationContext context)
             }
             return RangeFactory.all();
         } catch (IllegalStateException e) {
-            context.reportBadDefinition(handledType(), e.getMessage());
+            return context.reportBadDefinition(getValueType(context), e.getMessage());
+        }
+    }
+
+    private Range<?> deserializeRangeFromString(DeserializationContext context, JsonParser p)
+            throws IOException
+    {
+        String rangeInterval = p.getText();
+
+        if (rangeInterval.isEmpty()) {
             return null;
         }
+
+        if (isValidBracketNotation(rangeInterval)) {
```

Comment:
Tatu,

Thanks a lot for bringing this to my attention. I added validation to check for valid bracket characters.